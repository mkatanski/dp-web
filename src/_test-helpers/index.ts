export * from "./withResponsiveContext";
export * from "./withThemeProvider";
