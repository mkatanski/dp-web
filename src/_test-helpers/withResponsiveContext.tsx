import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
