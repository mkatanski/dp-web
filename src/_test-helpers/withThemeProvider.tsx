import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
