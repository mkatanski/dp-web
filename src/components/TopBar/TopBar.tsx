import React, { useContext } from "react";
import { SidebarToggleContext } from "lib/sidebar";
import {
  AppBar as MuiAppBar,
  AppBarProps,
  Toolbar as MuiToolbar,
  ToolbarProps,
  IconButton
} from "@material-ui/core";
import { MenuOutlined } from "@material-ui/icons";
import { Breakpoints } from "config/breakpoints";
import { useMediaQuery } from "react-responsive";
import styled from "styled-components";

const AppBar = styled(MuiAppBar)`
  box-shadow: none;
`;

const Toolbar = styled(MuiToolbar)`
  justify-content: flex-start;

  .menuButton {
    margin-right: ${({ theme }) => `${theme.spacing(1)}px`};
  }
`;

export type TopBarProps = AppBarProps & ToolbarProps;

export const TopBar: React.FC<TopBarProps> = ({ children }: TopBarProps) => {
  const toggleSidebar = useContext(SidebarToggleContext);
  const isTablet = useMediaQuery({ maxWidth: Breakpoints.largeDesktop });

  return (
    <AppBar>
      <Toolbar>
        {isTablet && (
          <IconButton
            onClick={toggleSidebar}
            className="menuButton"
            color="inherit"
          >
            <MenuOutlined />
          </IconButton>
        )}
        {children}
      </Toolbar>
    </AppBar>
  );
};
