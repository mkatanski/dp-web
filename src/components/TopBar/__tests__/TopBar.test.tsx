/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

import { withThemeProvider, withTabletSize } from "_test-helpers";
import { TopBar } from "../TopBar";

describe("components", () => {
  describe("TopBar", () => {
    it("should render TopBar component with children", () => {
      const tree = TestRenderer.create(
        withThemeProvider(<TopBar>children</TopBar>)
      );
      expect(tree).toMatchSnapshot();
    });

    it("should render TopBar component with tablet size", () => {
      const tree = TestRenderer.create(
        withTabletSize(withThemeProvider(<TopBar>children</TopBar>))
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
