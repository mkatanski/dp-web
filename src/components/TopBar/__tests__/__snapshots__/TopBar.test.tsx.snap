// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TopBar should render TopBar component with children 1`] = `
<div
  data-props={
    Object {
      "className": "sc-AxjAm hWpfU",
    }
  }
  id="MUI.AppBar"
>
  <div
    data-props={
      Object {
        "className": "sc-AxirZ iYKkTj",
      }
    }
    id="MUI.Toolbar"
  >
    children
  </div>
</div>
`;

exports[`components TopBar should render TopBar component with tablet size 1`] = `
<div
  data-props={
    Object {
      "className": "sc-AxjAm hWpfU",
    }
  }
  id="MUI.AppBar"
>
  <div
    data-props={
      Object {
        "className": "sc-AxirZ iYKkTj",
      }
    }
    id="MUI.Toolbar"
  >
    <span
      className="menuButton"
      color="inherit"
      id="MUI.IconButton"
      onClick={[Function]}
    >
      <span
        data-props={Object {}}
        id="MUI.Icon.MenuOutlined"
      />
    </span>
    children
  </div>
</div>
`;
